source/server/config_validation/server.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+
+#include "envoy/common/optional.h"
+#include "envoy/server/drain_manager.h"
+#include "envoy/server/instance.h"
+#include "envoy/ssl/context_manager.h"
+#include "envoy/tracing/http_tracer.h"
+
+#include "common/access_log/access_log_manager_impl.h"
+#include "common/runtime/runtime_impl.h"
+#include "common/ssl/context_manager_impl.h"
+#include "common/thread_local/thread_local_impl.h"
+
+#include "server/config_validation/api.h"
+#include "server/config_validation/cluster_manager.h"
+#include "server/config_validation/connection_handler.h"
+#include "server/config_validation/dns.h"
+#include "server/http/admin.h"
+#include "server/server.h"
+#include "server/worker.h"
+
+namespace Server {
+
+/**
+ * ValidationInstance does the bulk of the work for config-validation runs of Envoy. It implements
+ * Server::Instance, but some functionality not needed until serving time, such as updating
+ * health-check status, is not implemented. Everything else is written in terms of other
+ * validation-specific interface implementations, with the end result that we can load and
+ * initialize a configuration, skipping any steps that affect the outside world (such as
+ * hot-restarting or connecting to upstream clusters) but otherwise exercising the entire startup
+ * flow.
+ *
+ * If we finish initialization, and reach the point where an ordinary Envoy run would begin serving
+ * requests, the validation is considered successful.
+ */
+class ValidationInstance : Logger::Loggable<Logger::Id::main>, public Instance {
```

Comment:
Does this need to inherit from the implementation, or can it just be a bare minimum implementation that exports out methods similar to what is used in https://github.com/lyft/envoy/blob/master/test/config_test/config_test.cc?